test/e2e/cmd/run/run.go

```diff
@@ -254,28 +255,47 @@ func isOcp3Cluster(h *helper) bool {
 
 func (h *helper) initTestSecrets() error {
 	h.testSecrets = map[string]string{}
+
+	c, err := vault.NewClient()
+	if err != nil {
+		return err
+	}
+
 	if h.testLicense != "" {
-		bytes, err := os.ReadFile(h.testLicense)
+		bytes, err := vault.ReadFile(c, vault.SecretFile{
+			Name:          h.testLicense,
+			Path:          "test-licenses",
+			FieldResolver: vault.LicensePubKeyPrefix("enterprise"),
+		})
 		if err != nil {
-			return fmt.Errorf("reading %v: %w", h.testLicense, err)
+			return err
```

Comment:
Looking at what errors may be returned from `ReadFile`, I think this should be a more specific wrapped error here for context.  Also the 2 other instances below.